bin.src/inspect_job.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+#
+# This file is part of ap_verify.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Print the contents of a persisted Job object.
+
+This script takes as arguments one or more Job .json files, and prints the
+top-level metadata and a summary of any measurements. It does not print
+metrics that don't have measurements (there are far too many) or
+specifications (which are not helpful for testing measurement code).
+"""
+
+import sys
+import json
+
+from lsst.verify import Job
+
+
+def is_measurement_metadata(key, metrics):
+    """Test whether a job-level metadata key is really measurement metadata.
+
+    Parameters
+    ----------
+    key : `str`
+        The key to test.
+    metrics : iterable of `lsst.verify.Name` or of `str`
+        The metrics recorded in the job.
+
+    Returns
+    -------
+    result : `bool`
+        `True` if ``key`` represents measurement metadata, `False` otherwise.
+    """
+    for metric in metrics:
+        if str(metric) in key:
+            return True
```

Comment:
I don't understand why `str(metric) in key` always means that `metric` is actually metadata.

Reply:
Everything with a `key` is metadata; this is testing whether it's specifically measurement metadata (any measurement-specific keys are prefixed by the name of their metric).

Is this something I need to clarify somehow? I assumed anybody looking at the code would be familiar with `Job`'s output conventions.

Reply:
I think saying the first part of your first sentence in a comment (or the docstring) is enough to clarify it.